File: src/main/java/com/ms/petopia/api/recruitnotice/domain/RecruitNotice.java
package com.ms.petopia.api.recruitnotice.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class RecruitNotice {

    private Long recruitNoticeId;
    private Long writerId; // 작성한 EVENT_ADMIN
    private Long fairId; // 행사당 1건(UK)
    private String title; // 공고 제목
    private String content; // 모집요강·지원자격·문의처 등
    private String imageUrl; // 공고 대표 이미지, 선택
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private LocalDateTime recruitDeadline; // 모집 마감일시, 조기마감 가능

}

File: frontend/src/api/recruitNotice.ts
import { apiClient, type ApiEnvelope } from "./client";

export type BoothSlotStatus = "AVAILABLE" | "PENDING" | "CONFIRMED";

export interface RecruitNoticeBoothSlot {
  boothSlotsId: number;
  slotNumber: string;
  price: number;
  status: BoothSlotStatus;
  /** CONFIRMED일 때만 채워짐 */
  businessName: string | null;
  posX: number;
  posY: number;
  width: number;
  height: number;
  hallId: number;
  hallName: string;
  floorPlanImageUrl: string | null;
}

export interface RecruitNotice {
  recruitNoticeId: number;
  fairId: number;
  title: string;
  content: string;
  imageUrl: string | null;
  recruitDeadline: string;
  updatedAt: string;
  /** 저장값이 아니라 조회 시점마다 서버가 계산: 마감 지났으면 true */
  closed: boolean;
  boothSlots: RecruitNoticeBoothSlot[];
}

export interface RecruitNoticeUpsertRequest {
  title: string;
  content: string;
  recruitDeadline: string;
  imageObjectKey?: string;
}

export interface RecruitNoticeUpsertResult {
  recruitNoticeId: number;
  fairId: number;
  title: string;
  recruitDeadline: string;
  updatedAt: string;
}

// 백엔드가 ApiResponse<T>로 감싸서 응답하므로(fair 도메인과 달리 DTO를 바로 안 줌) data만 꺼내서 돌려준다.
// 모집 공고 상세 + 부스슬롯 현황 조회
export async function getRecruitNotice(fairId: number): Promise<RecruitNotice> {
  const response = await apiClient.get<ApiEnvelope<RecruitNotice>>(`/api/fairs/${fairId}/recruit-detail`);
  return response.data;
}

// 모집 공고 작성/수정 (행사 담당자용)
export async function upsertRecruitNotice(fairId: number, payload: RecruitNoticeUpsertRequest): Promise<RecruitNoticeUpsertResult> {
  const response = await apiClient.put<ApiEnvelope<RecruitNoticeUpsertResult>>(`/api/fairs/${fairId}/recruit-notice`, payload);
  return response.data;
}